tests/unit/test_extractor.py

```diff
@@ -0,0 +1,172 @@
+import pytest
+import pandas as pd
+from cognite.client.data_classes import EventWrite
+from cdf_fabric_replicator.extractor import CdfFabricExtractor
+from cdf_fabric_replicator.config import Config
+from cognite.extractorutils.base import CancellationToken
+from cognite.extractorutils.statestore import LocalStateStore
+from cognite.extractorutils.configtools import load_yaml
+
+@pytest.fixture(scope="session")
+def config_raw():
+    yield """
+    # Logging configuration
+    logger:
+        console:
+            level: INFO
+
+    # Cognite project to stream your datapoints from
+    cognite:
+        host: https://api.cognitedata.com
+        project: unit_test_extractor
+
+        idp-authentication:
+            token-url: https://get-a-token.com/token
+            client-id: abc123
+            secret: def567
+            scopes:
+                - https://api.cognitedata.com/.default
+        extraction-pipeline:
+            external-id: test-fabric-extractor
+
+    #Extractor config
+    extractor:
+        state-store:
+            local:
+                path: test_states.json
+        subscription-batch-size: 10000
+        ingest-batch-size: 100000
+        poll-time: 5
+
+    source:
+        abfss_event_table_path: source_table_path
+        abfss_directory: file_path
+        abfss_raw_time_series_table_path: /table/path
+        data_set_id: 123456
+
+    destination:
+        type: test_events
+        time_series_prefix: test_ts_
+    """
+@pytest.fixture(scope="session")
+def config(config_raw):
+    yield load_yaml(config_raw, Config)
+
+
+@pytest.fixture(scope="session")
+def extractor(config):
+    stop_event = CancellationToken()
+    extractor = CdfFabricExtractor(stop_event=stop_event)
+    extractor.config = config
+    extractor.client = extractor.config.cognite.get_cognite_client("test_extractor")
+    extractor.state_store = LocalStateStore(extractor.config.extractor.state_store.local.path)
```

Comment:
Can you explain why you have to define a Local state store instead of using a Mock?  I figure that it doesn't work but I'd like to understand why

Reply:
from what I remember I think it had to do with how that extractor is created and we needed a state store on creation and couldn't really mock everything that satisfied the `CDFFabricExtractor` class and its base class `Extractor` requirements